conda_libmamba_solver/solver.py

```diff
@@ -884,8 +884,16 @@ def _check_spec_compat(self, spec: MatchSpec | None) -> MatchSpec | None:
         for field in spec.FIELD_NAMES:
             value = spec.get_raw_value(field)
             if value:
-                if field == "channel" and str(value) == "<unknown>":
-                    continue
+                if field == "channel":
+                    if str(value) == "<unknown>":
+                        continue
+                    if not value.name:
```

Comment:
Is protection against `value` not having a name attributed needed?
```suggestion
                    if not getattr(value, "name", ""):
```

Reply:
It's guaranteed to be a `channel`, isn't it? Or maybe it could also be a string? Uuugh, types. Let's add it just in case.